packages/components/src/messages/components/reactions-dialog.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMemo } from 'react'
+
+import { Stack } from '@tamagui/web'
+
+import { Dialog } from '../../dialog'
+import { REACTIONS_ICONS } from '../../react-button/react-button'
+import { Tabs } from '../../tabs'
+import { UserList } from '../../user-list'
+
+import type { USerListProps } from '../../user-list'
+import type { ReactionsType, ReactionType } from '../types'
+
+type Props = {
+  initialReactionType: ReactionType
+  reactions: ReactionsType
+}
+
+export const ReactionsDialog = (props: Props) => {
+  const { initialReactionType, reactions } = props
+
+  const users: USerListProps['users'] = useMemo(() => {
```

Comment:
Why useMemo if it's static array of objects?

Reply:
That's true. Just expect that when we will be migrating to live data, there will be some filtering to show it correctly in the tabs, so this will need to be memoized.